Dev/ade7758

raezt opened the pull request:
## Pull Request Description

This is the initial commit for no-OS ADE7758 driver support and the accompanying example.
The driver implementation is a basic one with SPI communication and the example project uses the maxim platform.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

CiprianRegus commented on drivers/meter/ade7758/ade7758.c:
```diff
@@ -0,0 +1,958 @@
+		 int32_t *reg_data)
+{
+	int ret;
+	/* number of bytes = 2 for 8 bits register data */
```
It's up to you if you want to keep them, but these comments aren't really necessary

CiprianRegus commented on drivers/meter/ade7758/ade7758.c:
```diff
@@ -0,0 +1,958 @@
+		if (ret)
+			return ret;
+		if (IS_7BITS_REG(reg_addr))
+			*reg_data = no_os_sign_extend32(no_os_field_prep(NO_OS_GENMASK(6, 0),
```
I don't think you need to apply the [6:0] mask.

raezt replied:
True but I would leave it like so for consistency.

CiprianRegus commented on projects/eval-ade7758/src/platform/platform.c:
```diff
@@ -0,0 +1,113 @@
+/***************************************************************************//**
```
This file seems to be specific to the Maxim platform. I'd say we need to have a `maxim` subdirectory.

raezt commented:
V2 changelog (all comments addressed)

- removed the unnecessary comments
- added a new maxim subfolder in the platform folder and moved the platform files.

buha commented on drivers/meter/ade7758/ade7758.h:
```diff
@@ -0,0 +1,515 @@
+#define ADE7758_RESET_DEL					1
+
+/* ENABLE and DISABLE */
+#define ENABLE							1u
```
These are going to name clash with some library pretty soon, delete them and just use 0 and 1 inline wherever you use them.

buha commented on drivers/meter/ade7758/ade7758.h:
```diff
@@ -0,0 +1,515 @@
+ * @enum ade7758_ch_full_scale_sel_e
+ * @brief ADE7758 Current input full-scale select.
+ */
+enum ade7758_ch_full_scale_sel_e {
```
having _e suffix is nice probably but inconsistent with the rest of no-OS code so idk...

amiclaus commented on drivers/meter/ade7758/ade7758.c:
```diff
@@ -0,0 +1,958 @@
+			goto error_spi;
+		timeout--;
+	} while ((!reg_val) && (timeout > 0));
+	if (timeout <= 0) {
```
is there a case in which timeout will be < 0? I think this code block can be optimized a bit:
```
/* wait for reset ack */
do {
	ret = ade7758_get_int_status(dev, ADE7758_RESET_MSK, &reg_val);
	if (ret)
		goto error_spi;
} while ((!reg_val) && (timeout--));
if (!timeout) {
	ret = -ETIMEDOUT;
	goto error_spi;
}
```

amiclaus commented on drivers/meter/ade7758/ade7758.c:
```diff
@@ -0,0 +1,958 @@
+{
+	int ret;
+	/* register value */
+	uint32_t reg_val;
```
where is this used?

amiclaus commented on projects/eval-ade7758/builds.json:
```diff
@@ -0,0 +1,7 @@
+        "flags" : "TARGET=max32690"
+      }
+    }
+  }
```
github doesn't like not having a new line at eof :)

amiclaus reviewed (commented):
some comments on my side.

CiprianRegus commented on projects/eval-ade7758/src/common/common_data.c:
```diff
@@ -0,0 +1,156 @@
+	struct ade7758_energy_values energy_vals;
+	struct ade7758_rms_values rms_vals;
+	struct ade7758_freq_per_value freq_val;
+	float current_rms_val, voltage_rms_val;
```
This is project code, so I don't think it's that much of an issue...

raezt replied:
I would like to keep the float types in the example project.
